File: src/main/java/net/kingdommc/darkages/numinoustreasury/command/stamina/StaminaCommand.java
package net.kingdommc.darkages.numinoustreasury.command.stamina;

import net.kingdommc.darkages.numinoustreasury.NuminousTreasury;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static net.md_5.bungee.api.ChatColor.RED;

public final class StaminaCommand implements CommandExecutor, TabCompleter {

    private final StaminaViewCommand staminaViewCommand;
    private final StaminaSetCommand staminaSetCommand;

    private final List<String> viewAliases = List.of("view", "show");
    private final List<String> setAliases = List.of("set");
    private final List<String> subcommands = new ArrayList<>(){{
        addAll(viewAliases);
        addAll(setAliases);
    }};

    public StaminaCommand(NuminousTreasury plugin) {
        staminaViewCommand = new StaminaViewCommand(plugin);
        staminaSetCommand = new StaminaSetCommand(plugin);
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (args.length < 1) {
            sender.sendMessage(RED + "Usage: /stamina [set|view]");
            return true;
        }
        if (viewAliases.contains(args[0].toLowerCase())) {
            return staminaViewCommand.onCommand(sender, command, label, Arrays.stream(args).skip(1).toArray(String[]::new));
        } else if (setAliases.contains(args[0].toLowerCase())) {
            return staminaSetCommand.onCommand(sender, command, label, Arrays.stream(args).skip(1).toArray(String[]::new));
        } else {
            sender.sendMessage(RED + "Usage: /stamina [set|view]");
            return true;
        }
    }

    @Override
    public List<String> onTabComplete(CommandSender sender, Command command, String label, String[] args) {
        if (args.length < 1) {
            return subcommands;
        } else if (args.length == 1) {
            return subcommands.stream().filter(subcommand -> subcommand.startsWith(args[0].toLowerCase())).toList();
        } else {
            if (viewAliases.contains(args[0].toLowerCase())) {
                return staminaViewCommand.onTabComplete(sender, command, label, Arrays.stream(args).skip(1).toArray(String[]::new));
            } else if (setAliases.contains(args[0].toLowerCase())) {
                return staminaSetCommand.onTabComplete(sender, command, label, Arrays.stream(args).skip(1).toArray(String[]::new));
            } else {
                return List.of();
            }
        }
    }
}

File: src/main/kotlin/net/arvandor/numinoustreasury/command/stamina/StaminaCommand.kt
package net.arvandor.numinoustreasury.command.stamina

import net.arvandor.numinoustreasury.NuminousTreasury
import net.md_5.bungee.api.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter

class StaminaCommand(plugin: NuminousTreasury) : CommandExecutor, TabCompleter {
    private val staminaViewCommand = StaminaViewCommand(plugin)
    private val staminaSetCommand = StaminaSetCommand(plugin)

    private val viewAliases = listOf("view", "show")
    private val setAliases = listOf("set")
    private val subcommands = viewAliases + setAliases

    override fun onCommand(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>,
    ): Boolean {
        if (args.size < 1) {
            sender.sendMessage(ChatColor.RED.toString() + "Usage: /stamina [set|view]")
            return true
        }
        when (args[0].lowercase()) {
            in viewAliases -> {
                return staminaViewCommand.onCommand(
                    sender,
                    command,
                    label,
                    args.drop(1).toTypedArray(),
                )
            }
            in setAliases -> {
                return staminaSetCommand.onCommand(
                    sender,
                    command,
                    label,
                    args.drop(1).toTypedArray(),
                )
            }
            else -> {
                sender.sendMessage(ChatColor.RED.toString() + "Usage: /stamina [set|view]")
                return true
            }
        }
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>,
    ): List<String>? {
        return when {
            args.isEmpty() -> subcommands
            args.size == 1 ->
                subcommands
                    .filter { subcommand -> subcommand.startsWith(args[0].lowercase()) }
            else ->
                when (args[0].lowercase()) {
                    in viewAliases -> {
                        staminaViewCommand.onTabComplete(
                            sender,
                            command,
                            label,
                            args.drop(1).toTypedArray(),
                        )
                    }
                    in setAliases -> {
                        staminaSetCommand.onTabComplete(
                            sender,
                            command,
                            label,
                            args.drop(1).toTypedArray(),
                        )
                    }
                    else -> {
                        listOf()
                    }
                }
        }
    }
}
